### ISSUE TITLE
Responsive display

### ISSUE THREAD

**Issue by jcaillabet** [CoderLine/alphaTab#96]
Hi,

Is there a way to have a responsive rendering ? I'm new to alphaTab and I'm currently using it on a website with the same design for desktop and smartphone version, so I need the rendered context to react to different screen sizes, and screen size change. So I would need : 
- dynamic settings depending on screen size ( settings.Layout.AdditionalSettings['barsPerRow'], settings.Layout.AdditionalSettings['count'], settings.Width ... )
- refreshing the rendered context when screen change

_( I am doing modifications in my fork, so if it can be interesting I will make a pull-request )_

**Comment by Danielku15** [comment]
Hello. 

There is currently no automated way of letting alphaTab resize dyanamically with respect to the available width. I'm always investgating how to improve the layout engine more to achieve the fully dynamic sizing but there are still some obstacles.  

But: There is of course a way of rerender alphaTab manually to fit the desired space. As you already mentioned there are various settings that can be adjusted to control the layouting. First of all there is the most important setting: `AlphaTab.Settings.Width` With this setting you can control the width of the final music sheet being rendered in case of the page layout as shown in the [layouts demo](http://demo.alphatab.net/features.html#layouts). 

The `barsPerRow`setting might also be useful to adjust the default spacing to fit the available width better. 

To achieve a rerendering is quite easy when accessing the `AlphaTab.Rendering.ScoreRenderer` directly. 

```javascript
// load renderer and settings
var renderer = $('#alphaTab').alphaTab('renderer');
var settings = renderer.Settings;
// adjust some properties and rerender
settings.Width = 500;
renderer.Invalidate();
```

This means if you listen to particular DOM events for window resizing you can adjust the settings of the renderer and make the music sheet fit your new layout.  

Depending on the client performance you would see some flickering since there is no out-of-the-box "double buffering". If you take a look at the [API wrapper](https://github.com/CoderLine/alphaTab/blob/master/Source/AlphaTab.JavaScript/Platform/JavaScript/JsApiBase.cs#L47) you can see how the ScoreRenderer is connected to the DOM. 

AlphaTab renders every line of music as a so called partial. Once layouting and rendering is done the PartialRenderFinished will be triggered. In case of HTML5 rendering this will be a canvas element that can be added to the DOM, for SVG the raw SVG string is returned. This can be clearly seen when using web-workers off-screen for rendering as demonstrated in the [Showcase demo](http://demo.alphatab.net/features.html#all). 

I hope this infos make it a bit more clear how to achieve a dynamic layouting as you need it. 
.

**Comment by jcaillabet** [comment]
Thank you,

Finally I could make a responsive behaviour by listening window resizing event. I Don't seems to have flickering, maybe my page is slow to refresh when I change screen size but it is not a big problem.

( Sometime I had to make change in alphaTab code when i chose a scale different than 1 before rendering. Now I discarded those changes and there is no more errors as I made better my initialization, so finally I suppose there is no problem to change the scale before the rendering )

I'm really interested to improve alphaTab to suit my needs, so if it can be interesting I will think how to include my dynamic change of settings, for exemple adding an item in the **Settings** to tell we want a automatic display.
